kolibri/core/assets/src/core-app/mediator.js

```diff
@@ -321,4 +336,52 @@ module.exports = class Mediator {
   off(...args) {
     this._eventDispatcher.$off(...args);
   }
+
+  _fetchLanguageAssets(moduleName, language) {
+    let promise;
+    if (this._languageAssetRegistry[moduleName] &&
+      this._languageAssetRegistry[moduleName][language] &&
+      this._languageAssetRegistry[moduleName][language].promise) {
+      promise = this._languageAssetRegistry[moduleName][language].promise;
+    } else {
+      promise = new Promise((resolve, reject) => {
```

Comment:
A promise allows you to arbitrarily chain callbacks from the Promise, so you can pass around the promise and let many people reference it, without them having to know about what else anyone is doing with it.